client/lib/sites-list/list.js

```diff
@@ -372,7 +372,7 @@ SitesList.prototype.getSelectedSite = function() {
  * @api private
  */
 SitesList.prototype.setSelectedSite = function( siteID ) {
-	if ( ! siteID ) {
+	if ( ! siteID || this.selected === siteID ) {
```

Comment:
Every navigation inside My Sites such as Stats, Plan, Posts, etc. trigger `my-sites.controller.siteSelection` which in turn set the selected `siteID` if one is found in the current path.
This check eliminates unnecessary `change` event that will effectively do nothing.

I can refactor this change to another PR as the change to `MySitesSidebar` is not dependent on it.